src/test/java/com/mongodb/jdbc/MongoSQLStatementTest.java

```diff
@@ -62,75 +71,71 @@ void testNoop(MongoSQLConnectionTest.TestInterface ti) throws SQLException {
 
     @Test
     void testExecuteQueryEmptyResult() throws SQLException {
-        // TODO: SQL-535, requires MongoResultSet implementation
-        //        AtomicInteger rowCnt = new AtomicInteger();
-        //
-        //        when(mongoCursor.hasNext()).thenAnswer(invocation -> rowCnt.get() < 1);
-        //
-        //        when(mongoCursor.next())
-        //                .thenAnswer(
-        //                        invocation -> {
-        //                            rowCnt.incrementAndGet();
-        //                            return generateMetadataDoc();
-        //                        });
-        //
-        //        ResultSet rs = mongoStatement.executeQuery("select * from foo");
-        //        ResultSetMetaData metaData = rs.getMetaData();
-        //        assertEquals(2, metaData.getColumnCount());
-        //        // TODO: add metadata check after ResultSetMetaData code is merged
-        //        // need to call next() first
-        //        assertThrows(
-        //                SQLException.class,
-        //                new Executable() {
-        //                    @Override
-        //                    public void execute() throws Throwable {
-        //                        rs.getInt(1);
-        //                    }
-        //                });
-        //        assertFalse(rs.next());
-        //        assertThrows(
-        //                SQLException.class,
-        //                new Executable() {
-        //                    @Override
-        //                    public void execute() throws Throwable {
-        //                        rs.getInt(1);
-        //                    }
-        //                });
-        //        assertTrue(rs.isLast());
+        AtomicInteger rowCnt = new AtomicInteger();
```

Comment:
Why does this have to be an AtomicInteger? I don't see where is the concurrency coming from, do you know?

Reply:
I'm not sure why it needs to be `Atomic` either, but I assume this decision was made because something failed. Not really worried in a test file.